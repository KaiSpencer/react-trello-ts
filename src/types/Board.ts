import {CSSProperties} from 'react'

export interface BoardData {
  lanes: Lane[]
}

export interface Lane {
  id: string
  title?: string
  label?: string
  style?: CSSProperties
  cards: Card[]
  currentPage?: number
  droppable?: boolean
  labelStyle?: CSSProperties
  cardStyle?: CSSProperties
  disallowAddingCard?: boolean
  [key: string]: any
}

export interface Card {
  id: string
  title?: string
  label?: string
  description?: string
  laneId?: string
  style?: CSSProperties
  draggable?: boolean
  [key: string]: any
}
