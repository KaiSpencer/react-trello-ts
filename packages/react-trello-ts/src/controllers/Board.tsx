import React, { createContext, FC, PropsWithChildren } from "react";
import classNames from "classnames";
import uuidv1 from "uuid/v1";
import { BoardContainer } from "./BoardContainer";
import * as DefaultComponets from "../components";
import { BoardData } from "../types/Board";
import { store } from "../store/store";

export const Board: FC<
	PropsWithChildren<{
		id?: string;
		className?: string;
		components: typeof DefaultComponets;
		data: BoardData;
		t?: any;
	}>
> = ({ data, children, className, components, id, t, ...rest }) => {
	const allClassNames = classNames("react-trello-board", className || "");

	return (
		<BoardContext.Provider value={store}>
			<components.GlobalStyle />
			<BoardContainer
				id={id || uuidv1()}
				data={data}
				components={components}
				t={t}
				className={allClassNames}
				{...rest}
			/>
		</BoardContext.Provider>
	);
};

export const BoardContext = createContext(store);
