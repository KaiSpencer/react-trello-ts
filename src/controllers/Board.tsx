import classNames from "classnames";
import React, { FC, PropsWithChildren, createContext } from "react";
import { v1 } from "uuid";
import * as DefaultComponets from "../components";
import { store } from "../store/store";
import { BoardData } from "../types/Board";
import { BoardContainer } from "./BoardContainer";

export const Board: FC<
	PropsWithChildren<{
		id?: string;
		className?: string;
		components: typeof DefaultComponets;
		data: BoardData;
		t?: any;
	}>
> = ({ data, children, className, components, id, t, ...rest }) => {
	const allClassNames = classNames("react-trello-board", className || "");

	return (
		<BoardContext.Provider value={store}>
			{/*
			// @ts-ignore */}
			<components.GlobalStyle />
			<BoardContainer
				id={id || v1()}
				data={data}
				components={components}
				t={t}
				className={allClassNames}
				{...rest}
			/>
		</BoardContext.Provider>
	);
};

export const BoardContext = createContext(store);
