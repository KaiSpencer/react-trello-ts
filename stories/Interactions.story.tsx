import React from 'react'
import {storiesOf} from '@storybook/react'

import Board from '../src'

const data = {
  lanes: [
    {
      id: 'lane1',
      title: 'Planned Tasks',
      cards: [
        {id: 'Card1', title: 'Card1', description: 'foo card', metadata: {id: 'Card1'}},
        {id: 'Card2', title: 'Card2', description: 'bar card', metadata: {id: 'Card2'}}
      ]
    },
    {
      id: 'lane2',
      title: 'Executing',
      cards: [{id: 'Card3', title: 'Card3', description: 'foobar card', metadata: {id: 'Card3'}}]
    }
  ]
}

storiesOf('Advanced Features', module).add(
  'Event Handling',
  () => (
    <Board
      draggable
      data={data}
      onCardClick={(cardId, metadata, laneId) =>
        alert(`Card with id:${cardId} clicked. Has metadata.id: ${metadata.id}. Card in lane: ${laneId}`)
      }
      onLaneClick={laneId => alert(`Lane with id:${laneId} clicked`)}
    />
  ),
  {info: 'Adding event handlers to cards'}
)
