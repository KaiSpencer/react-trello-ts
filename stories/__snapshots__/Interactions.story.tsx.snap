// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Advanced Features EventHandling smoke-test 1`] = `
<div draggable="false"
     class="sc-iGgWBj iSNfA"
>
  <div>
    <div class="smooth-dnd-container horizontal">
      <div class="smooth-dnd-draggable-wrapper">
        <section draggable="false"
                 class="sc-kAyceB dgWEjS react-trello-lane"
        >
          <header class="sc-gsFSXq sc-imWYAI imRIDz gLxuAH">
            <span class="sc-kpDqfm gDkBAH">
              Planned Tasks
            </span>
          </header>
          <div class="sc-dhKdcB ikKiDc">
            <div class="smooth-dnd-container vertical">
              <div class="smooth-dnd-draggable-wrapper">
                <article data-id="Card1"
                         class="sc-jlZhew sc-cwHptR clGdKf fQNcVl react-trello-card"
                >
                  <header class="sc-gsFSXq sc-jEACwC imRIDz jNaWkr">
                    <span draggable="true"
                          class="sc-kpDqfm sc-cPiKLX cFicSq cnDDQK"
                    >
                      Card1
                    </span>
                    <span class="sc-dAlyuH sc-dLMFU iHiWje dvkFIZ">
                    </span>
                    <div class="sc-hmdomO gJuJJJ">
                      <button class="sc-jsJBEP djrAfI">
                        ✖
                      </button>
                    </div>
                  </header>
                  <div class="sc-eDPEul cRhiim">
                    foo card
                  </div>
                </article>
              </div>
              <div class="smooth-dnd-draggable-wrapper">
                <article data-id="Card2"
                         class="sc-jlZhew sc-cwHptR clGdKf fQNcVl react-trello-card"
                >
                  <header class="sc-gsFSXq sc-jEACwC imRIDz jNaWkr">
                    <span draggable="true"
                          class="sc-kpDqfm sc-cPiKLX cFicSq cnDDQK"
                    >
                      Card2
                    </span>
                    <span class="sc-dAlyuH sc-dLMFU iHiWje dvkFIZ">
                    </span>
                    <div class="sc-hmdomO gJuJJJ">
                      <button class="sc-jsJBEP djrAfI">
                        ✖
                      </button>
                    </div>
                  </header>
                  <div class="sc-eDPEul cRhiim">
                    bar card
                  </div>
                </article>
              </div>
            </div>
          </div>
        </section>
      </div>
      <div class="smooth-dnd-draggable-wrapper">
        <section draggable="false"
                 class="sc-kAyceB dgWEjS react-trello-lane"
        >
          <header class="sc-gsFSXq sc-imWYAI imRIDz gLxuAH">
            <span class="sc-kpDqfm gDkBAH">
              Executing
            </span>
          </header>
          <div class="sc-dhKdcB ikKiDc">
            <div class="smooth-dnd-container vertical">
              <div class="smooth-dnd-draggable-wrapper">
                <article data-id="Card3"
                         class="sc-jlZhew sc-cwHptR clGdKf fQNcVl react-trello-card"
                >
                  <header class="sc-gsFSXq sc-jEACwC imRIDz jNaWkr">
                    <span draggable="true"
                          class="sc-kpDqfm sc-cPiKLX cFicSq cnDDQK"
                    >
                      Card3
                    </span>
                    <span class="sc-dAlyuH sc-dLMFU iHiWje dvkFIZ">
                    </span>
                    <div class="sc-hmdomO gJuJJJ">
                      <button class="sc-jsJBEP djrAfI">
                        ✖
                      </button>
                    </div>
                  </header>
                  <div class="sc-eDPEul cRhiim">
                    foobar card
                  </div>
                </article>
              </div>
            </div>
          </div>
        </section>
      </div>
    </div>
  </div>
</div>
`;
